### PR TITLE
Update support for casdoor 

### PR DESCRIPTION
**Pull request by jakiuncle**
Motivation
Update support for casdoor, because casdoor has updated.

Fixes #446 

Master Issue: #446 

 Modifications

I make it connect casdoor in new version.

### CODE REVIEW

**Comment by hsluoyz** [comment]
@tuteng plz review

File: front-end/src/router/index.js [hunk 1]
```diff
@@ -77,7 +77,7 @@ export const constantRouterMap = [
 ]
 
 export default new Router({
-  // mode: 'history', // require service support
+  mode: 'history', // require service support
```
Comment (tuteng) [inline front-end/src/router/index.js:80]: I remember that this change may cause packaging failed, can you confirm @urfreespace

**Comment by nodece** [comment]
Any updates?

**Comment by jakiuncle** [comment]
i will try it

**Comment by jakiuncle** [comment]
> Any updates?
Because the version of casdoor is updated, the original code can no longer take effect
So i update it

File: front-end/src/router/index.js (hunk 1 above)
Comment (urfreespace) [reply front-end/src/router/index.js:80]: yeah, this will cause the route jumper to fail

Comment (jakiuncle) [reply front-end/src/router/index.js:80]: Ok, i will not use this mode

Comment (jakiuncle) [reply front-end/src/router/index.js]: Hi,it not use mode:'history' now.

**Comment by jakiuncle** [comment]
@tuteng plz review

File: src/main/java/org/apache/pulsar/manager/controller/LoginController.java [hunk 2]
```diff
@@ -2,9 +2,9 @@
  * Licensed under the Apache License, Version 2.0 (the "License");
  * you may not use this file except in compliance with the License.
  * You may obtain a copy of the License at
- *
- *     http://www.apache.org/licenses/LICENSE-2.0
- *
+ * <p>
```
Comment (nodece) [inline src/main/java/org/apache/pulsar/manager/controller/LoginController.java:5]: Revert this.

File: src/main/java/org/apache/pulsar/manager/controller/LoginController.java [hunk 3]
```diff
@@ -128,30 +122,40 @@ public ResponseEntity<Map<String, Object>> logout() {
...
-    })
-    @RequestMapping(value = "/casdoor", method =  RequestMethod.POST)
-    public ResponseEntity<Map<String, Object>> callback(
-            @RequestBody Map<String, String> body) {
-        Map<String, Object> result = Maps.newHashMap();
+    @ApiOperation(value = "casdoor Login pulsar manager")
+    @ApiResponses({@ApiResponse(code = 200, message = "ok"), @ApiResponse(code = 500, message = "Internal server error")})
+    @RequestMapping(value = "/casdoor", method = RequestMethod.POST)
+    public ResponseEntity<Map<String, Object>> callback(@RequestBody Map<String, String> body) {
+        String userAccount, userPassword, token;
```
Comment (nodece) [inline src/main/java/org/apache/pulsar/manager/controller/LoginController.java:129]:
```suggestion
     
```

File: src/main/java/org/apache/pulsar/manager/controller/LoginController.java [hunk 4]
```diff
@@ -128,30 +122,40 @@ public ResponseEntity<Map<String, Object>> logout() {
...
+        UserInfoEntity userInfoEntity = new UserInfoEntity();
+        if (!userInfoEntityOptional.isPresent()) {
+            userInfoEntity.setUserId(0);
+            userInfoEntity.setName(casdoorUser.getName());
+            userInfoEntity.setPassword(casdoorUser.getPassword());
+            userInfoEntity.setExpire(0);
+            usersRepository.save(userInfoEntity);
+        } else {
+            userInfoEntity = userInfoEntityOptional.get();
+        }
```
Comment (nodece) [inline src/main/java/org/apache/pulsar/manager/controller/LoginController.java:145]:
```suggestion
        }
           String userAccount, userPassword, token;
```

File: src/main/java/org/apache/pulsar/manager/controller/LoginController.java [hunk 5]
```diff
@@ -128,30 +122,40 @@ public ResponseEntity<Map<String, Object>> logout() {
...
+        userAccount = casdoorUser.getName();
+        userPassword = casdoorUser.getPassword();
+        token = jwtService.toToken(userAccount + userPassword + System.currentTimeMillis());
+        userInfoEntity.setAccessToken(token);
+        result.put("login", "success");
```
Comment (nodece) [inline src/main/java/org/apache/pulsar/manager/controller/LoginController.java:150]:
```suggestion
        Map<String, Object> result = Maps.newHashMap();
        result.put("login", "success");
```

File: src/main/java/org/apache/pulsar/manager/controller/LoginController.java [hunk 6]
```diff
@@ -128,30 +122,40 @@ public ResponseEntity<Map<String, Object>> logout() {
...
         String state = body.get("state");
-        String token = casdoorAuthService.getOAuthToken(code, state);
-        if(!token.startsWith("error")){
-            result.put("error", token.substring(7));
-            return ResponseEntity.ok(result);
-        }
-        result.put("login", "success");
+        String casdoortoken = casdoorAuthService.getOAuthToken(code, state);
+        CasdoorUser casdoorUser = casdoorAuthService.parseJwtToken(casdoortoken);
+        Map<String, Object> result = Maps.newHashMap();
```
Comment (nodece) [inline src/main/java/org/apache/pulsar/manager/controller/LoginController.java:132]:
```suggestion
```

File: src/main/java/org/apache/pulsar/manager/controller/LoginController.java [hunk 7]
```diff
@@ -128,30 +122,40 @@ public ResponseEntity<Map<String, Object>> logout() {
...
+        } else {
+            userInfoEntity = userInfoEntityOptional.get();
+        }
+        userAccount = casdoorUser.getName();
+        userPassword = casdoorUser.getPassword();
+        token = jwtService.toToken(userAccount + userPassword + System.currentTimeMillis());
+        userInfoEntity.setAccessToken(token);
+        result.put("login", "success");
+        usersRepository.update(userInfoEntity);
+        headers.add("token", token);
```
Comment (nodece) [inline src/main/java/org/apache/pulsar/manager/controller/LoginController.java:154]:
```suggestion
        HttpHeaders headers = new HttpHeaders();
        headers.add("token", token);
```

File: src/main/java/org/apache/pulsar/manager/controller/LoginController.java [hunk 8]
```diff
@@ -128,30 +122,40 @@ public ResponseEntity<Map<String, Object>> logout() {
...
-        String token = casdoorAuthService.getOAuthToken(code, state);
-        if(!token.startsWith("error")){
-            result.put("error", token.substring(7));
-            return ResponseEntity.ok(result);
-        }
-        result.put("login", "success");
+        String casdoortoken = casdoorAuthService.getOAuthToken(code, state);
+        CasdoorUser casdoorUser = casdoorAuthService.parseJwtToken(casdoortoken);
+        Map<String, Object> result = Maps.newHashMap();
         HttpHeaders headers = new HttpHeaders();
```
Comment (nodece) [inline src/main/java/org/apache/pulsar/manager/controller/LoginController.java:132]:
```suggestion

```

**Review by nodece — CHANGES_REQUESTED** [review]
Request changes.

**Review by nodece — APPROVED** [review]

**Comment by jakiuncle** [comment]
@tuteng @urfreespace plz review

**Comment by jakiuncle** [comment]
@tuteng @urfreespace plz review

**Comment by hsluoyz** [comment]
@rdhabalia @michaeljmarshall @eolivelli plz review

**Review by tuteng — APPROVED** [review]

**Pull request merged**